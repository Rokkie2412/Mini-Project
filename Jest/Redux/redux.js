export const ADD = 'ADD'
export const DELETE = 'DELETE'

export const addArray = (data) =>({
    type:ADD,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch(action.type){
        case ADD:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE:
            return{
                ...state,arr:state.arr.filter(arr => arr.id !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer