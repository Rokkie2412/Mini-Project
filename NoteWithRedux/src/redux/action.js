import { ADDARRAY } from "./actionType";

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})