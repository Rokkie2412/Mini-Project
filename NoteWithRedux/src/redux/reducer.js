import { ADDARRAY } from "./actionType";

const initialState = {
    arr:[],
    
};

export const mainReducer = (state = initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
           
        default:
            return state;
    }
}