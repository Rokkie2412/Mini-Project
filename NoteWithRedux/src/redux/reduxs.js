import remove from "lodash.remove"
export const ADDARRAY = 'ADDARRAY'
export const DELETENOTE = 'DELETENOTE'
export const EDITNOTE = 'EDITNOTE'

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})

export const deleteFromArray = (title) => ({
    type:DELETENOTE,
    payload:title
})
export const EditNote = (title) => ({
    type:EDITNOTE,
    payload:title,
})



const initialState = []

export const mainReducer = (state=initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return[
                ...state,action.payload
            ]
        case DELETENOTE:
            const deleteNewArray = remove(state, obj => {
                return obj.title != action.payload
            })
            return deleteNewArray
        case EDITNOTE:
            // const filterDelete = map(state,obj =>{
            //     return obj.title == action.payload
            // })
            const filtered = state.filter((state,obj)=>{
                if(obj.title === action.payload){
                    var tamp = obj.title
                    console.log(tamp);
                }
            })
        default:
            return state;
    }
}

export default mainReducer