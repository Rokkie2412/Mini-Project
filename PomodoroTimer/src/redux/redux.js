
/**
 * @format
 * @flow
 */
export const EXCHAHNGETOSECONDS : string = 'EXCHANGETOSECOND'
export const EXCHAHNGETOSECONDBREAK :string = 'EXCHANGETOSECONDBREAK'
export const EXCHAHNGETOSECONDLONGBREAK :string = 'EXCHANGETOSECONDLONGBREAK'
export const GETDATA:string = 'GETDATA'

export const exchangetosecwork = (hour:string,minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDS,
    payloadhourwork : hour,
    payloadminutework : minute,
    payloadsecondwork : second
})

export const exchangetosecBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDBREAK,
    payloadminutebreak : minute,
    payloadsecondbreak : second
})

export const exchangetosecLongBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDLONGBREAK,
    payloadminutelongbreak : minute,
    payloadsecondlongbreak : second
})

let initialState : Object  ={
    resultWorkinSecond : 0,
    resultBreakinSecond : 0,
    resultLongBreakinSecond : 0
}

export const mainReducer = (state:Object=initialState, action:function):Function =>{
    switch (action.type) {
        case EXCHAHNGETOSECONDS:
            
            const hourTosecond : number = action.payloadhourwork * 3600
            const minuteTosecond : number = action.payloadminutework * 60
            const res : number = hourTosecond+minuteTosecond+action.payloadsecondwork
            return {
                ...state,
                resultWorkinSecond:res
            }
        case EXCHAHNGETOSECONDBREAK:
            const minuteTosecondBreak : number= action.payloadminutebreak * 60
            const resBreak : number = minuteTosecondBreak+action.payloadsecondbreak
            return {
                ...state,
                resultBreakinSecond:resBreak
            }
        case EXCHAHNGETOSECONDLONGBREAK:
            const minuteToSecondLongBreak : number= action.payloadminutelongbreak * 60
            const resLongBreak : number = minuteToSecondLongBreak+action.payloadsecondlongbreak
            return{
                ...state,
                resultLongBreakinSecond:resLongBreak
            }
        default:
            return state
    }
}

export default mainReducer