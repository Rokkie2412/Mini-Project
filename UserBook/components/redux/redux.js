export const ADD_CONTACT = 'ADD_CONTACT'
export const SHOW_CONTACT = 'SHOW_CONTACT'
export const EDITCONTACT = 'EDITCONTACT'

const ApiUrl = 'https://simple-contact-crud.herokuapp.com/contact'


export const addContact = (firstName,lastName,age,image) =>({
    type:ADD_CONTACT,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

export const editContact = (itemId,firstName,lastName,age,image) => ({
    type:EDITCONTACT,
    payloadID : itemId,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

let initialState = {
    arr:[],
}

export const mainReducer = (state=initialState, action) => {
    switch (action.type) {
        case ADD_CONTACT:
            const postContact = () =>{
                fetch('https://simple-contact-crud.herokuapp.com/contact',{
                    method:'POST',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(postContact())
        case EDITCONTACT :
            const editContact = () =>{
                const tempid = action.payloadID 
                
                fetch(`https://simple-contact-crud.herokuapp.com/contact/${tempid}`,{
                    method:'PUT',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                 }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(editContact())
        default:
            return state
    }
}