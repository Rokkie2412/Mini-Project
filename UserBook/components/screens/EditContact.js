import React,{useState} from 'react'
import {Text,TextInput,Pressable,View,StyleSheet,Image,Alert} from 'react-native'
import Ion from 'react-native-vector-icons/Ionicons'
import Feather from 'react-native-vector-icons/Feather'
import { useDispatch } from "react-redux";
import { editContact } from '../redux/redux';
import { launchCamera, launchImageLibrary} from 'react-native-image-picker';
import { showMessage } from 'react-native-flash-message';
// import { Sae, Akira } from 'react-native-textinput-effects'
const EditContact = ({navigation,route}) => {

    //mengambil data yang di passd dari screen Home
    let itemID = route.params.itemID
    let TempFirstName = route.params.firstName
    let TempLastname = route.params.lastName
    let TempAge = route.params.age
    let StringAge = "" + TempAge
    let TempPhoto = route.params.photo
    const [Error,setError] = useState('')
    const [errorMessage,setErrorMessage] = useState('')

    // console.log(itemID, TempFirstName,TempLastname,TempAge,TempPhoto)
    const dispatch = useDispatch()
    const [firstName,setFirstName] = useState(TempFirstName)
    const [LastName,setLastName] = useState(TempLastname)
    const [Age,setAge] = useState(StringAge)
    const [imageCamera,setImageCamera] = useState(TempPhoto)
    
    //melakukan disptach ke redux yang berisi POST data ke API
    const contactEdit = () =>{
        dispatch(editContact(itemID,firstName,LastName,Age,imageCamera))
    }

    //code untuk membuka camera
    const openCamera = () => {
        
        const option={
            mediaType : 'photo',
            quality : 0.7
        }
        launchCamera(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Pick Image From Photo');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImageCamera(data.uri)
                console.log(data)
            }
        })
    }
    //code untuk membuka gallery
    const openGallery = () => {
        
        const option={
            mediaType : 'photo',
            quality : 0.7
        }
        launchImageLibrary(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Pick Image From Gallery');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImageCamera(data.uri)
                console.log(data)
            }
        })
    }

    const checkValidate = () =>{
        if(firstName.length < 3 || LastName.length < 3 || Age === ""){
                        showMessage({
                        message:'Edit Failed',
                        description: errorMessage,
                        animationDuration:600,
                        floating:true,
                        type:'warning',
                        icon:'warning'
                        })
                    }else{
                        contactEdit()
                        navigation.navigate('Home')
                    }
    }

    const vaidatename = (newFirst) =>{
        if(newFirst.length < 3){
                            setError('First name and last name length atleast must have 3 character')
                            setErrorMessage('Last name length atleast mush have 3 character')
                            
                        }else{
                            setErrorMessage('')
                            setError('') 
                        }
    }
    
    return(
        <View style={styles.mainContainer}>
            <View style={styles.header}>
                <Pressable testID='backButton' onPress={()=>{
                    navigation.navigate('Home')
                }}>
                    <Feather style={styles.backIcon} name='x'/>
                </Pressable>
                <Text style={styles.titleheader}>Edit contact</Text>
                <Pressable
                testID='submitButton'
                onPress={()=>{
                    checkValidate()
                }}
                >
                    <Ion style={styles.completeIcon} name='checkmark-sharp'/>
                </Pressable>
            </View>
            <View style={styles.photoContainer}>
                <Pressable
                onPress={()=>{
                     Alert.alert(
                        'Choose Media for pick an image',
                        'Camera or Gallery',[
                            {
                                text:'Cancel',
                                onPress: ()=> console.log('Cancel'),
                            },
                            {
                                text:'Camera',
                                onPress: ()=>openCamera(),
                            },
                            {
                                text:'Gallery',
                                onPress: ()=>openGallery(),
                                styles:'cancel'
                            }
                        ]
                    )
                }}
                >
                    {imageCamera === 'N/A' ? <Ion style={styles.person} name='person'/> :
                    <Image source={{uri:imageCamera}} style={styles.imagePhoto}/>
                    }
                </Pressable>
            </View>
            <View style={styles.InputContainer}>
            {/* <Sae    
                    labelStyle={{color:'#2C3333'}}
                    inputStyle={{color:'#2C3333'}}
                    label='First Name'
                    iconClass={Ion}
                    iconName={'pencil'}
                    iconColor='#2C3333'
                    inputPadding={16}
                    labelHeight={24}
                    borderHeight={2}
                    utoCapitalize={'none'}
                    autoCorrect={false}
                    value={firstName}
                    onChangeText={(newFirst)=>{
                        setFirstName(newFirst)
                        {if(newFirst.length < 3){
                            setError('First name and last name length atleast must have 3 character')
                            setErrorMessage('First name length atleast mush have 3 character')
                           
                        }else{
                            setErrorMessage('')
                            setError('') 
                        }}
                    }}
                    style={styles.Lastname}
                />
                <Sae
                    labelStyle={{color:'#2C3333'}}
                    inputStyle={{color:'#2C3333'}}
                    label='Last Name'
                    iconClass={Ion}
                    iconName={'pencil'}
                    iconColor='#2C3333'
                    inputPadding={16}
                    labelHeight={24}
                    borderHeight={2}
                    utoCapitalize={'none'}
                    autoCorrect={false}
                    value={LastName}
                    onChangeText={(newLast)=>{
                        setLastName(newLast)
                        {if(newLast.length < 3){
                            setError('First name and last name length atleast must have 3 character')
                            setErrorMessage('Last name length atleast mush have 3 character')
                            
                        }else{
                            setErrorMessage('')
                            setError('') 
                        }}
                    }}
                    style={styles.Lastname}
                /> */}
               <TextInput
                    testID='firstname'
                    value={firstName}
                    onChangeText={(newFirst)=>{
                        setFirstName(newFirst)
                        {if(newFirst.length < 3){
                            setError('First name and last name length atleast must have 3 character')
                            setErrorMessage('Last name length atleast mush have 3 character')
                            
                        }else{
                            setErrorMessage('')
                            setError('') 
                        }}
                    }}
                    placeholder='Input Last Name'
                    autoCapitalize={'none'}
                    autoCorrect={false}
                    style={styles.Lastname}
                />
                <TextInput
                    value={LastName}
                    onChangeText={(newLast)=>{
                        setLastName(newLast)
                        {if(newLast.length < 3){
                            setError('First name and last name length atleast must have 3 character')
                            setErrorMessage('Last name length atleast mush have 3 character')
                            
                        }else{
                            setErrorMessage('')
                            setError('') 
                        }}
                    }}
                    placeholder='Input Last Name'
                    autoCapitalize={'none'}
                    autoCorrect={false}
                    style={styles.Lastname}
                />
                <Text testID='error' style={{marginHorizontal:32,fontSize:11,color:'black',marginVertical
                :7,}}>{Error}</Text>
                {/* <Akira
                    keyboardType='numeric'
                    label='Age'
                    labelStyle={{color:'#2C3333',bottom:3}}
                    borderColor='#2C3333'
                    labelHeight={30}
                    value={Age}
                    onChangeText={(newAge)=>{
                        setAge(newAge)
                        if(newAge === ''){
                            setErrorMessage('Age must not be empty')
                    }
                    }}
                    style={styles.age}
                /> */}
                <TextInput
                    value={Age}
                    onChangeText={(newAge)=>{
                        setAge(newAge)
                        if(newAge === ''){
                            setErrorMessage('Age must not be empty')
                    }
                    }}
                    style={styles.age}
                    placeholder='Age?'
                    keyboardType = 'numeric'
                />
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    mainContainer:{
        flex:1,
        backgroundColor:'#fff'
    },
    header:{
        justifyContent:'space-between',
        flexDirection:'row',
        margin:5,
    },
    backIcon:{
        padding:10,
        fontSize:30,
        color:'#2C3333',
        borderRadius:40,
    },
    titleheader:{
        fontSize:20,
        fontWeight:'600',
        padding:10,
        color:'#2C3333',
    },
    completeIcon:{
        padding:10,
        fontSize:30,
        color:'#2C3333',
        borderRadius:40,
    },
    person:{
        backgroundColor:'#EEEEEF',
        color:'white',
        fontSize:40,
        padding:20,
        borderRadius:100,
        
    },
    photoContainer:{
        flexDirection:'row',
        alignContent:'center',
        justifyContent:'center',
    },
    InputContainer:{
        flex:1
    },
Firstname:{
        backgroundColor:'#EFEFEE',
        borderRadius:50,
        marginTop:15,
        marginHorizontal:12,
        fontWeight:'bold',
        paddingLeft:20,
    },
Lastname:{
        color:'#2C3333',
        marginTop:8,
        marginHorizontal:35,
        fontWeight:'bold',
        paddingLeft:20,
        borderBottomWidth:1,
        borderColor:'#333',
    },
    age:{
        marginTop:8,
        marginHorizontal:12,
        fontWeight:'bold',
        width:100,
        alignSelf:'center',
        textAlign:'center',
        borderBottomWidth:1,
        borderColor:'#333',
    },
    imagePhoto:{
        height:90,
        width:90,
        borderRadius:50,
    }

})

export default EditContact