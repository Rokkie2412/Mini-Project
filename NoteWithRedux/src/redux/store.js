import { createStore } from "redux";
import {mainReducer} from "./reduxs";

export const store = createStore(mainReducer)