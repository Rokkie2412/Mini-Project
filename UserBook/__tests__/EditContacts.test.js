import React from "react";
import EditContact from "../components/screens/EditContact";
import { Provider } from "react-redux";
import { mainReducer } from "../components/redux/redux";
import { createStore } from "redux";
import { create,act } from "react-test-renderer";

jest.mock(
  'react-native-vector-icons/Ionicons',
  () => 'Mocked Ionicons',
);

jest.mock(
  'react-native-vector-icons/Feather',
  () => 'Mocked Ionicons',
);

jest.mock(
    'react-native-flash-message',
    ()=> 'Mocked react-native-flash-message'
)

jest.mock(
    'react-native-image-picker',
    ()=> 'Mocked react-native-image-picker'
)

jest.mock(
    'react-native-textinput-effects',
    ()=>'Mocked react native effect input'
)

const navigation={
    navigate:jest.fn()
}

const store = createStore(mainReducer,{status:'default status'})

const mockedparams = {
            route:{params : {itemID:'route.params.itemID'}},
            navigate:jest.fn()
        }

const tree = create(
        <Provider store={store}>
            <EditContact navigation={navigation} {...mockedparams}/>
        </Provider>
        )

describe('Testing Edit Contact snaps',()=>{
    test('render corretly',()=>{
        mockedparams
        tree
        expect(tree).toMatchSnapshot();
    })
})


test('navigation to Home',()=>{
    const button = tree.root.findByProps({testID:'backButton'}).props
    button.onPress()
    expect(navigation.navigate).toBeCalledWith('Home')
})

// test('navigation to Home',()=>{
//     const button = tree.root.findByProps({testID:'submitButton'}).props
//     button.onPress()
//     expect(navigation.navigate).toBeCalledWith('Home')
// })