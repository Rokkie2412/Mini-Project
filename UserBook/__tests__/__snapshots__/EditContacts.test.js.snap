// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Edit Contact snaps render corretly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 5,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="backButton"
    >
      <Mocked Ionicons
        name="x"
        style={
          Object {
            "borderRadius": 40,
            "color": "#2C3333",
            "fontSize": 30,
            "padding": 10,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#2C3333",
          "fontSize": 20,
          "fontWeight": "600",
          "padding": 10,
        }
      }
    >
      Edit contact
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="submitButton"
    >
      <Mocked Ionicons
        name="checkmark-sharp"
        style={
          Object {
            "borderRadius": 40,
            "color": "#2C3333",
            "fontSize": 30,
            "padding": 10,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 50,
            "height": 90,
            "width": 90,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Input Last Name"
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#333",
          "color": "#2C3333",
          "fontWeight": "bold",
          "marginHorizontal": 35,
          "marginTop": 8,
          "paddingLeft": 20,
        }
      }
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Input Last Name"
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#333",
          "color": "#2C3333",
          "fontWeight": "bold",
          "marginHorizontal": 35,
          "marginTop": 8,
          "paddingLeft": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 11,
          "marginHorizontal": 32,
          "marginVertical": 7,
        }
      }
    >
      
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Age?"
      style={
        Object {
          "alignSelf": "center",
          "borderBottomWidth": 1,
          "borderColor": "#333",
          "fontWeight": "bold",
          "marginHorizontal": 12,
          "marginTop": 8,
          "textAlign": "center",
          "width": 100,
        }
      }
      value="undefined"
    />
  </View>
</View>
`;
